### ISSUE TITLE
Impossible to partially merge semantics

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#12401]
Imagine a contact list where you can click on a list entry to call the contact or you can click on a trailing `i` icon to bring up more information about that contact. Essentially, a tile of that contact list would look like this:
```
----------------------------------
|                     --------   |
| Max Mustermann      |   i  |   |
|                     --------   |
----------------------------------
```

Only the big and the little box should be a11y focusable. When the big box is focused, it should read out "Max Mustermann". Essentially, we would want the SemanticsTree to look like this:

```
SemanticsNode(label: "Max Mustermann", actions: [SemanticsAction.tap])
   - SemanticsNode(label: "Details", actions: [SemanticsAction.tap])
```

With our current implementation, that SemanticsTree is very hard (impossible?) to achieve because we don't allow partial merging.

Out of the box, the SemanticsTree would currently look like this:

```
SemanticsNode(actions: [SemanticsAction.tap])
   - SemanticsNode(label: "Max Mustermann")
   - SemanticsNode(label: "Details", actions: [SemanticsAction.tap])
```

All three Nodes are focusable ==> sad.

All you can currently really do about it is throw in a `MergeSemantics`, which yields the following SemanticsTree:

```
SemanticsNode(label: "Max Mustermann\nDetails", actions: [SemanticsAction.tap])
```

It all gets munched into one node, you can't focus on the little `i` anymore and you can only perform one action (the call action probably wins) ==> double sad.

What we really need is some way to partially merge semantics. Ideally, the system should figure out automatically what needs to be merged without the developer telling flutter.

**Comment by goderbauer** [comment]
The new a11y algorithm has solved this.
